api/provider/aws/common.go

```diff
@@ -7,8 +7,48 @@ import (
 	"github.com/aws/aws-sdk-go/service/ec2"
 	"github.com/aws/aws-sdk-go/service/ec2/ec2iface"
 	"github.com/quintilesims/layer0/api/tag"
+	"github.com/quintilesims/layer0/common/errors"
 )
 
+func lookupEntityEnvironmentID(store tag.Store, entityType, entityID string) (string, error) {
+	tags, err := store.SelectByTypeAndID(entityType, entityID)
+	if err != nil {
+		return "", err
+	}
+
```

Comment:
This is what I meant by my earlier comment:
```
tags, err := store.SelectByTypeAndID(...)
...

// if there are 0 tags for the specified type and id, I think it's reasonable to assume that entity does not exist
if len(tags) == 0 {
    // we will need a new function in common/errors to match 'entityType' to an error code, 
    // e.g. "environment" = errors.EnvironmentDoesNotExist
    return "", errors.NewEntityDoesNotExistError(entityType, entityID)
}

if tag, ok := tags.WithKey(...)

// we don't really have a meaningful error code  for this scenario (other than `UnexpectedError`), 
// so just use a basic error here
return "", fmt.Errorf("Could not resolve ...")
```

Reply:
Aaaaah, I see - so both scenarios are useful here. :+1:

Reply:
✅